cli/internal/core/engine.go

```diff
@@ -180,14 +186,29 @@ func (e *Engine) Prepare(options *EngineBuildingOptions) error {
 
 					return err
 				}
-
+				// delete Taskanem if it was found
+				delete(missing, taskName)
 				traversalQueue = append(traversalQueue, taskID)
 			}
 		}
 	}
 
 	visited := make(util.Set)
 
+	// validate that all tasks passed were found
+	missingList := []string{}
+	for t := range missing {
+		missingList = append(missingList, t)
+	}
+	sort.Strings(missingList)
+
+	if len(missingList) > 0 {
+		if len(missingList) == 1 {
+			return fmt.Errorf("Could not find \"%s\" in project", missingList[0])
+		}
+		return fmt.Errorf("Could not find the following tasks in project: %s", strings.Join(missingList, ", "))
+	}
```

Comment:
I appreciate the intent here but I'd prefer that the error message be more-easily searchable as opposed to being 100% grammatically correct.

```suggestion
	if len(missingList) > 0 {
		return fmt.Errorf("Could not find the following tasks in project: %s", strings.Join(missingList, ", "))
	}
```